src/Examine/Search/IFieldSelectableQuery.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Examine.Search
+{
+    public interface IFieldSelectableQuery : IQuery
+    {
+        /// <summary>
+        /// Return only the specified fields. Use <see cref="SelectFields(ISet{string})"/> when possible as internally a new HashSet is created on each call
+        /// </summary>
+        /// <param name="fieldNames">The field names for fields to load</param>
+        /// <returns></returns>
+        IBooleanOperation SelectFields(params string[] fieldNames);
+
+        /// <summary>
+        /// Return only the specified fields
+        /// </summary>
+        /// <param name="fieldNames">The field names for fields to load</param>
+        /// <returns></returns>
+        IBooleanOperation SelectFields(ISet<string> fieldNames);
+
+        /// <summary>
+        /// Return only the specified field. Use <see cref="SelectFields(ISet{string})"/> when possible as internally a new HashSet is created on each call
+        /// </summary>
+        /// <param name="fieldNames">The field name of the field to load</param>
+        /// <returns></returns>
+        IBooleanOperation SelectField(string fieldName);
+
+        /// <summary>
+        /// Return only the specified fields Use <see cref="SelectFields(ISet{string})"/> when possible as internally a new HashSet is created on each call
+        /// </summary>
+        /// <remarks>The Id field will also be retrieved as it is a required field.</remarks>
+        /// <param name="fieldNames">The field names for fields to load. Key should be the field name, value should be null</param>
+        /// <returns></returns>
+        [Obsolete("Use SelectFields(ISet<string> fieldNames) to reduce allocations")]
```

Comment:
Just noticed this Obsolete declaration on new code, was this left behind during refactoring?

Reply:
It's for anyone migrating from v0.9 as they should use Hashtable in v0.9, but ISet<string> in 1.0

Reply:
Ah I see, ok i'll need to see how this plays out with the 0.9 stuff.